src/test/java/SentinelTesting.java

```diff
@@ -0,0 +1,242 @@
+
+				        if (val == -1) {
+				            System.out.println("ERROR: str is null");
+				        }
+				    }
+				}
+				        """;
+		String expectedOutput = """
+				public class SentinelTest {
+				    public void test() {
+				        String str = "Hello World";
+				        int val = 0;
+
+				        if (str == null) {
+				            val = -1;
+				        }
+
+				        if (str == null) {
+				            System.out.println("ERROR: str is null");
+				        }
+				    }
+				}
+				                        """;
+		test(input, expectedOutput);
+	}
+
+	@Test
+	public void swappedSignsTest() {
+		String input = """
+				public class SentinelTest {
+				    public void test() {
+				        String str = "Hello World";
+				        int val = 0;
+
+				        if (str == null) {
+				            val = -1;
+				        }
+
+				        if (val != -1) {
+				            System.out.println("ERROR: str is null");
+				        }
+				    }
+				}
+				        """;
+		String expectedOutput = """
+				public class SentinelTest {
+				    public void test() {
+				        String str = "Hello World";
+				        int val = 0;
+
+				        if (str == null) {
+				            val = -1;
+				        }
+
+				        if (str != null) {
+				            System.out.println("ERROR: str is null");
+				        }
+				    }
+				}
+				                        """;
+		test(input, expectedOutput);
+	}
+
+	@Test
+	public void inverseCheck() {
+		String input = """
+				public class SentinelTest {
+				    public void test() {
+				        String str = "Hello World";
+				        int val = 0;
+
+				        if (str != null) {
+				            val = 1;
+				        }
+
+				        if (val == 1) {
+				            System.out.println("Str is not null");
+				        }
+				    }
+				}
+				        """;
+		String expectedOutput = """
+				public class SentinelTest {
+				    public void test() {
+				        String str = "Hello World";
+				        int val = 0;
+
+				        if (str != null) {
+				            val = 1;
+				        }
+
+				        if (str != null) {
+				            System.out.println("Str is not null");
+				        }
+				    }
+				}
+				                        """;
+		test(input, expectedOutput);
+	}
+
+	@Test
+	public void indirectCheck() {
+		String input = """
+				public class SentinelTest {
+				    public void test() {
+				        String str = "Hello World";
+				        int val = 0;
+
+				        if (str == null) {
+				            val = -1;
+				        }
+
+				        if (val == 0) {
+				            System.out.println("Str is not null");
+				        }
+				    }
+				}
+				        """;
+		String expectedOutput = """
+				public class SentinelTest {
+				    public void test() {
+				        String str = "Hello World";
+				        int val = 0;
+
+				        if (str == null) {
+				            val = -1;
+				        }
+
+				        if (str != null) {
+				            System.out.println("Str is not null");
```

Comment:
Can you create a version of this test that shows that the refactoring isn't applied when line 150 (`val = -1`) is replaced with `val = 0` (i.e., the same value that `val` originally had)?

Reply:
This is not correct.`val` being 0 carries no information about whether `str` is null, and no refactoring should be applied: whether or not we enter the `if` statement, `val` is zero afterwards. Please add a test case that shows that this code doesn't apply a refactoring in this case, as I originally asked for.